nocodb/nocodb
# qrcode column type MVP

## Change Summary

MVP for adding a QR code column type

Contributes to https://github.com/nocodb/nocodb/issues/3581

### Scope
* Ability to Add and Edit QR code columns
  * Supports the following column types for the referenced value to be encoded in the QR (since these seem to not have formatting ambiguities - besides potentially formula? -, which reduces the risk to have unwanted ambiguities in the QR codes): 
    * SingleTextLine
    * LongText
    * PhoneNumber
    * Formula
    * URL
    * Email

## Next Steps in this PR
- [x] Proper supporting for referencing formula column type
- [x] Fix "QR Code not re-rendering after column (e.g. formula or the qr code column itself) or a cell value referenced by the QR code column gets updated"
- [x] Adding some basic unit tests
- [x] Adding some basic playwright tests

## Change type


## Test/ Verification

* Create a column of the supported types (see above) 
* Create a QR Code column and set the referenced column to the one you just created
* Scan the QR codes with your phone and check that the values are as expected
* Try out clicking on the QR code (should open up an larger popup version of it) 
* Change one cell value of the referenced column and check that the QR code in the same row is auto updating
* Also check other view types (like Kanban and Gallery) that the QR code is shown correctly

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of
